src/main/java/de/tum/in/www1/artemis/service/FeedbackService.java

```diff
@@ -148,6 +147,21 @@ public Feedback createFeedbackFromTestCase(String testName, List<String> testErr
      * @return A filtered and better formatted error message
      */
     private static String processResultErrorMessage(final ProgrammingLanguage programmingLanguage, final String message) {
+        final String timeoutDetailText = "The test case execution timed out. This indicates issues in your code such as endless for / while loops or issues with recursion. Please carefully review your code to avoid such issues. In case you are absolutely sure that there are no issues like this, please contact your instructor to check the setup of the test.";
+        // Overwrite timeout exception messages for Junit4, Junit5 and other
+        List<String> exceptions = Arrays.asList("org.junit.runners.model.TestTimedOutException", "java.util.concurrent.TimeoutException",
+                "org.awaitility.core.ConditionTimeoutException", "Timed?OutException");
+        Pattern findPattern = Pattern.compile("^.*(" + String.join("|", exceptions) + ").*");
+        Matcher matcher = findPattern.matcher(message);
+        if (matcher.find()) {
+            return matcher.group(1) + ": " + timeoutDetailText;
```

Comment:
it's unclear what matcher.group(1) is here. Please extract a variable with a descriptive name.
We should only remove the exception name. The text after the exception might still contain useful information.

Reply:
I would add the timeoutDetailText first and then in a new line the original exception text (without the exception name, so e.g. without  `org.awaitility.core.ConditionTimeoutException`)